SampleApplications/Samples/XamarinClient/XamarinClient/SampleClient.cs

```diff
@@ -0,0 +1,469 @@
+                        }
+
+                        ViewNode viewNode = currentNode as ViewNode;
+
+                        if (viewNode != null)
+                        {
+                            currentNodeEventNotifier = viewNode.EventNotifier;
+                        }
+
+                        MethodNode methodNode = currentNode as MethodNode;
+
+                        if (methodNode != null )
+                        {
+                            currentNodeExecutable = methodNode.UserExecutable;
+                        }
+
+                        browserTree.currentView.Add(new ListNode()
+                        {
+                            id = node,
+                            NodeName = currentNode.DisplayName.Text.ToString(),
+                            nodeClass = currentNode.NodeClass.ToString(),
+                            accessLevel = currentNodeAccessLevel.ToString(),
+                            eventNotifier = currentNodeEventNotifier.ToString(),
+                            executable = currentNodeExecutable.ToString(),
+                            children = false,
+                            ImageUrl = null
+                        });
+                    }
+                }
+                return browserTree;
+            }
+            catch (Exception exception)
+            {
+                Disconnect(session);
+                return null;
+            }
+        }
+
+        public string VariableRead(string node)
+        {
+            try
+            {
+                DataValueCollection values = null;
+                DiagnosticInfoCollection diagnosticInfos = null;
+                ReadValueIdCollection nodesToRead = new ReadValueIdCollection();
+                ReadValueId valueId = new ReadValueId();
+                valueId.NodeId = new NodeId(node);
+                valueId.AttributeId = Attributes.Value;
+                valueId.IndexRange = null;
+                valueId.DataEncoding = null;
+                nodesToRead.Add(valueId);
+                ResponseHeader responseHeader = session.Read(null, 0, TimestampsToReturn.Both, nodesToRead, out values, out diagnosticInfos);
+                string value = "";
+                if (values[0].Value != null)
+                {
+                    var rawValue  = values[0].WrappedValue.ToString();
+                    value = rawValue.Replace("|","\r\n").Replace("{","").Replace("}","");
+                }
+                return value;
+            }
+            catch (Exception exception)
+            {
+                return null;
+            }
+        }
+
+        private void CertificateValidator_CertificateValidation(CertificateValidator validator, CertificateValidationEventArgs e)
+        {
+            e.Accept = (e.Error.StatusCode == StatusCodes.BadCertificateUntrusted);
+            // This will just accept any certificate
+            //e.Accept = true;
```

Comment:
dialog to accept?

Reply:
if not auto accept, open dialog to accept?

---
In reply to: [165611923](https://github.com/OPCFoundation/UA-.NETStandard/pull/337#discussion_r165611923) [](ancestors = 165611923)